bin/vault_utils/vault_interface.py

```diff
@@ -38,18 +38,20 @@ def _authenticate_approle(self):
             "auth/{0}/login".format(self._approle_path)
         )
 
-        r = requests.post(auth_url, data=auth_data)
+        headers = {}
+        self._add_namespace(headers)
+        r = requests.post(auth_url, data=auth_data, headers=headers)
+        r.raise_for_status()
 
         # TODO - this needs error handling
         return r.json()["auth"]["client_token"]
 
-    @property
-    def headers(self):
-        headers = { "X-Vault-Token": self._token }
+    def _add_namespace(self, headers):
         if self._namespace:
             headers["X-Vault-Namespace"] = self._namespace
 
-        return headers
+    def _add_token(self, headers):
+        headers["X-Vault-Token"] = self._token
```

Comment:
I separated the task of adding headers into adding the namespace and token headers separately, allowing them to be called as needed for the type of request being made.  Approle authentication only adds the namespace (approle credentials are separate), but everything afterwards adds both namespace and token.